src/app-layout/utils/use-drawers.ts

```diff
@@ -105,8 +105,12 @@ export function useDrawers(
   if (toolsDrawer && combinedDrawers.length > 0) {
     combinedDrawers.unshift(toolsDrawer);
   }
-  const activeDrawer = combinedDrawers.find(drawer => drawer.id === activeDrawerId);
-  const activeDrawerIdResolved = activeDrawer?.id; // only defined when corresponding drawer exists
+  // support toolsOpen in runtime-drawers-only mode
+  let activeDrawerIdResolved =
```

Comment:
non-blocking: 
Can we make the `activeDrawerIdResolved` as a `const` instead? we can reference the actual active drawer id using the `activeDrawer` variable already.

Down below we can just pass the id by returning:

```
return {
activeDrawerId: activeDrawer?,id,
{..rest
}
```

Reply:
There is a re-assignment on line 113. I could work around it by introducing intermediate variables

```js
// support toolsOpen
const activeDrawerIdWithTools = toolsProps.toolsOpen && (ownDrawers?.items ?? []).length === 0 ? TOOLS_DRAWER_ID : activeDrawerId
const activeDrawer = combinedDrawers.find(drawer => drawer.id === activeDrawerIdWithTools);
// ensure that the variable is only defined when the respective drawer exist
const activeDrawerIdResolved = activeDrawer?.id;
```

But then the naming gets confusing. I think having a pipeline like is acceptable code

```js
let foo = something();
foo = somethingElse(foo)
foo = anotherOne(foo)
```

If you have anything more to add, I am open for DM discussion